src/view/com/modals/SwitchAccount.tsx

```diff
@@ -32,11 +33,15 @@ function SwitchAccountCard({account}: {account: SessionAccount}) {
   const {data: profile} = useProfileQuery({did: account.did})
   const isCurrentAccount = account.did === currentAccount?.did
   const {onPressSwitchAccount} = useAccountSwitcher()
+  const closeAllActiveElements = useCloseAllActiveElements()
 
   const onPressSignout = React.useCallback(() => {
     track('Settings:SignOutButtonClicked')
-    logout()
-  }, [track, logout])
+    closeAllActiveElements()
+    setTimeout(() => {
```

Comment:
Curious why this `logout` needs to be inside a `setTimeout`?

Reply:
Ah good call, added comment. Without that, the modal never finishes closing and remains open. CC @gaearon